src/components/v5/common/CompletedAction/partials/StreamingPayment/partials/StreamingPaymentWidget/StreamingPaymentWidget.tsx

```diff
@@ -0,0 +1,286 @@
+import { ActionTypes } from '~redux';
+import { type ClaimStreamingPaymentPayload } from '~redux/sagas/expenditures/claimStreamingPayment.ts';
+import Numeral from '~shared/Numeral/Numeral.tsx';
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader.tsx';
+import { formatText } from '~utils/intl.ts';
+import getStreamingPaymentAmountsLeft from '~utils/streamingPayment.ts';
+import { getSelectedToken } from '~utils/tokens.ts';
+import { ACTION_TYPE_FIELD_NAME } from '~v5/common/ActionSidebar/consts.ts';
+import { useGetStreamingPaymentData } from '~v5/common/ActionSidebar/hooks/useGetStreamingPaymentData.ts';
+import StreamingPaymentStatusPill from '~v5/common/ActionSidebar/partials/StreamingPaymentStatusPill/StreamingPaymentStatusPill.tsx';
+import { StreamingPaymentStatus } from '~v5/common/ActionSidebar/partials/StreamingPaymentStatusPill/types.ts';
+import { getStatus } from '~v5/common/ActionSidebar/partials/StreamingPaymentStatusPill/utils.ts';
+import Button from '~v5/shared/Button/Button.tsx';
+import MenuWithSections from '~v5/shared/MenuWithSections/MenuWithSections.tsx';
+
+import FinalizeWithPermissionsInfo from '../FinalizeWithPermissionsInfo/FinalizeWithPermissionsInfo.tsx';
+
+import { useHasEnoughTokensToClaim } from './hooks.ts';
+import { type StreamingPaymentWidgetProps } from './types.ts';
+
+const displayName =
+  'v5.common.CompletedAction.partials.StreamingPayment.partials.StreamingPaymentWidget';
+
+const MSG = defineMessages({
+  streamingStatus: {
+    id: `${displayName}.streamingStatus`,
+    defaultMessage: 'Streaming status',
+  },
+  paidToDate: {
+    id: `${displayName}.paidToDate`,
+    defaultMessage: 'Paid to date',
+  },
+  availableToClaim: {
+    id: `${displayName}.availableToClaim`,
+    defaultMessage: 'Available to claim',
+  },
+  payFunds: {
+    id: `${displayName}.payFunds`,
+    defaultMessage: 'Pay funds',
+  },
+  insufficientFunds: {
+    id: `${displayName}.insufficientFunds`,
+    defaultMessage: 'Insufficient funds in team. Ensure team is funded.',
+  },
+  transferFunds: {
+    id: `${displayName}.transferFunds`,
+    defaultMessage: 'Transfer required funds',
+  },
+});
+
+const StreamingPaymentWidget: FC<StreamingPaymentWidgetProps> = ({
+  action,
+}) => {
+  const { loadingStreamingPayment, streamingPayment, refetchStreamingPayment } =
+    useGetStreamingPaymentData(action.expenditureId);
+  const { colony } = useColonyContext();
+  const { streamingPaymentsAddress } = useEnabledExtensions();
+  const { currentBlockTime: blockTime, refreshBlockTime } =
+    useCurrentBlockTime();
+  const {
+    actionSidebarToggle: [, { toggleOn: toggleActionSidebarOn }],
+  } = useActionSidebarContext();
+
+  const { amountAvailableToClaim, amountClaimedToDate } = useMemo(
+    () =>
+      getStreamingPaymentAmountsLeft(
+        streamingPayment,
+        Math.floor(blockTime ?? Date.now() / 1000),
+      ),
+    [blockTime, streamingPayment],
+  );
```

Comment:
Couldn't the existing hook be used?

Reply:
I don't think that it should be a hook. There's no functionality inside that requires this to be a hook. I need it to get the latest data every time `blockTime` is updated inside useEffect with counter to get the right `amountAvailableToClaim` and that's why it's in `useMemo`

Reply:
There are many other hooks that do not need to be hooks, yet we chose to make them such for consistency and readability reasons.

My concern is that we now have an identical fragment of logic in 2 different places and it's not clear which one to use going forward. If you intend to keep this as a function, please remove the hook

Reply:
Hook removed